src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,674 @@
+            asserts.assert_equal(sorted(attribute_ids), sorted(returned_attributes), "Expected attribute list doesn't match")
+
+        # Step 4
+
+        # TH sends the Read Request Message to the DUT to read a global attribute from all clusters at that Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint, Attribute = Specific Global Attribute]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(4, "Send Request Message to read one global attribute from all clusters at that endpoint")
+        attribute_path_1 = AttributePath(
+            EndpointId=0,
+            ClusterId=None,
+            AttributeId=global_attribute_ids.GlobalAttributeIds.ATTRIBUTE_LIST_ID)
+
+        read_request = await self.default_controller.ReadAttribute(
+            self.dut_node_id,
+            [attribute_path_1]
+        )
+
+        for endpoint in read_request:
+            asserts.assert_in(Clusters.Objects.Descriptor, read_request[endpoint].keys(), "Descriptor cluster not in output")
+            asserts.assert_in(Clusters.Objects.Descriptor.Attributes.AttributeList,
+                              read_request[endpoint][Clusters.Objects.Descriptor], "AttributeList not in output")
+        asserts.assert_equal(sorted(all_attributes[endpoint][Clusters.Descriptor]
+                             [Clusters.Descriptor.Attributes.ServerList]),
+                             sorted([x.id for x in read_request[endpoint]]))
+        # Step 5
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters on all Endpoints
+        ### AttributePath = [[]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the clusters to the DUT.
+        self.print_step(5, "Send Request Message to read all attributes from all clusters on all endpoints")
+        read_request = await self.default_controller.ReadAttribute(self.dut_node_id, [()])
+
+        # NOTE: This is checked in its entirety in IDM-10.1
+
+        parts_list_a = read_request[0][Clusters.Descriptor][Clusters.Descriptor.Attributes.PartsList]
+        parts_list_b = self.endpoints[0][Clusters.Descriptor][Clusters.Descriptor.Attributes.PartsList]
+        asserts.assert_equal(parts_list_a, parts_list_b, "Parts list is not the expected value")
+
+        for endpoint in read_request:
+            returned_clusters = sorted([x.id for x in read_request[endpoint]])
+            server_list = sorted(read_request[endpoint][Clusters.Descriptor][Clusters.Descriptor.Attributes.ServerList])
+            asserts.assert_equal(returned_clusters, server_list)
+
+        for endpoint in read_request:
+            for cluster in read_request[endpoint]:
+                # Endpoint 1 seems an issue with ModeSelect (ServerList has an extra 4293984257)
```

Comment:
This needs changes, but I think the easier solution is to just omit this step because full wildcard is properly and thoroughly checked in 10.1

Reply:
Commented for now; might remove completely